### PR TITLE
Add JSON schema build & format scripts

### PR DESCRIPTION
**Pull request by neftaly**
This PR adds prettier (to fix indents and newlines), and typescript-json-schema (to allow repeatable schema builds). This is intended to improve the maintainability of this package.

There are some inconsistencies between `schema.json` and the typescript def files, can you please have a look at my comments and see which I should fix and which I should leave?

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -136,14 +136,6 @@
           "description": "Direction from which the camera should view the object when zoomed. Default perspective and automatic rotation is used if zero."
         }
       },
-      "required": [
```
Comment (neftaly) [inline schema.json]: src/object-template.ts:135: `ZoomViewDirection?: Vector;` (optional)

File: schema.json [hunk 2]
```diff
@@ -388,7 +373,6 @@
           "type": "number"
         }
       },
-      "required": ["B", "G", "R"],
```
Comment (neftaly) [inline schema.json]: Color appears not to be required

File: schema.json [hunk 3]
```diff
@@ -662,7 +629,6 @@
           "type": "number"
         }
       },
-      "required": ["Name", "X", "Y", "Z"],
```
Comment (neftaly) [inline schema.json]: This *should* be required, right?

File: schema.json [hunk 4]
```diff
@@ -1365,7 +1324,6 @@
           "type": "string"
         }
       },
-      "required": ["Type"],
```
Comment (neftaly) [inline schema.json]: This *should* be required?

File: schema.json [hunk 5]
```diff
@@ -1381,7 +1339,6 @@
           "type": "number"
         }
       },
-      "required": ["X", "Y", "Z"],
```
Comment (neftaly) [inline schema.json]: This *should* be required?

**Comment by neftaly** [comment]
see 5b62e66699e4bfe5c2f9538e2479c2f3afcfe054 for changes to schema.json, some changes seem good, and some seem bad (i.e. some `required` fields should be kept). The goal is to make the typescript definitions the same as the compiled schema file, so that future change can be made & kept in sync.

**Comment by plasticity-studios** [comment]
Thanks for the PR! Adding `--required` to the parameters for `typescript-json-schema` will reintroduce all required fields. Why would you only want to keep some? Most (perhaps all, but I haven't tested it completely) of the required fields in the current schema.json will result in an error if missing.

Adding `ObjectTemplate` and `ObjectType` (by parsing all files instead of only `templates.ts`) doesn't seem to add value to the schema.json, because they are not used directly in valid JSON files.

**Comment by neftaly** [comment]
Thanks! I wasn't sure if you had manually added these fields or if they were generated by typescript-json-schema. I have updated the command & rebuilt.

**Pull request merged**